test/test_other.py

```diff
@@ -2691,35 +2672,53 @@ def test_dwarf_sourcemap_names(self):
       ([], False, False, False),
       (['-g0'], False, False, False),
       (['-g1'], False, False, False),
+      (['-g1', '-O2'], False, False, False),
+      (['--minify=0'], False, False, False),
       # last numeric g flag "wins", so g0 overrides -g
       (['-g', '-g0'], False, False, False),
       (['-g2'], False, False, True),
       (['-gline-tables-only'], True, False, True),
       (['--profiling'], False, False, True),
       (['--profiling-funcs'], False, False, True),
+      (['-O2', '--profiling-funcs'], False, False, True),
       (['-g'], True, False, True),
       (['-g3'], True, False, True),
-      (['-O2', '-g'], True, False, True),
+      (['-O1', '-g'], True, False, True),
+      (['-O3', '-g'], True, False, True),
       (['-gsplit-dwarf'], True, False, True),
+      # TODO: It seems odd that -gsource-map leaves behind a name section. Should it?
       (['-gsource-map'], False, True, True),
```

Comment:
Not related to this PR, but why does `-gsource-map` generate the name section?

Reply:
I think that might be because we just set the internal debug level to "debug" (3 or 4) when generating source maps. We could probably avoid that with some work.